vm/message/src/chain_message.rs

```diff
@@ -102,3 +102,69 @@ impl Cbor for ChainMessage {
         }
     }
 }
+
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use crate::{signed_message, unsigned_message};
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    /// Wrapper for serializing and deserializing a SignedMessage from JSON.
+    #[derive(Deserialize, Serialize)]
+    #[serde(transparent)]
+    pub struct ChainMessageJson(#[serde(with = "self")] pub ChainMessage);
+
+    /// Wrapper for serializing a SignedMessage reference to JSON.
+    #[derive(Serialize)]
+    #[serde(transparent)]
+    pub struct ChainMessageJsonRef<'a>(#[serde(with = "self")] pub &'a ChainMessage);
+
+    impl From<ChainMessageJson> for ChainMessage {
+        fn from(wrapper: ChainMessageJson) -> Self {
+            wrapper.0
+        }
+    }
+
+    impl From<ChainMessage> for ChainMessageJson {
+        fn from(msg: ChainMessage) -> Self {
+            ChainMessageJson(msg)
+        }
+    }
+
+    pub fn serialize<S>(m: &ChainMessage, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        enum ChainMessageSer {
+            #[serde(with = "unsigned_message::json")]
+            Unsigned(UnsignedMessage),
+            #[serde(with = "signed_message::json")]
+            Signed(SignedMessage),
+        };
+
+        let chain_message_ser = match m {
+            ChainMessage::Unsigned(s) => ChainMessageSer::Unsigned(s.clone()),
+            ChainMessage::Signed(s) => ChainMessageSer::Signed(s.clone()),
+        };
```

Comment:
you shouldn't need to clone to serialize, can change the variants in the Ser type to just hold a ref.

Side note, where are you getting this Json format? I can't find anywhere that matches the default enum serialization